java/com/google/turbine/parse/ConstExpressionParser.java

```diff
@@ -455,6 +452,9 @@ private long parseLong(String text, int radix) {
       }
       eat();
     }
+    if (token == Token.LBRACK) {
+        return finishClassLiteral(pos, asClassTy(pos, bits.build()));
+    }
```

Comment:
@cushon Moving this block below the `while` loop fixes the test. All other tests still pass.

Reply:
Thanks for investigating this and for the report. I started looking at in parallel and came to a similar conclusion.